### DISCUSSION TITLE
Question regarding tags and VNet format

### DISCUSSION THREAD

**Discussion by esdcharles** [Azure/ipam#95] [Q&A] [answered]
First off I want to say I am absolutely impressed with this IPAM offering and thanks to all who have been contributing to this discussion. I have deployed (full) the IPAM and have developed a test area in which I am putting together a PoC to which users can submit requests for IP reservations and we can centrally manage them within the Hub. I have read the discussions, docs and wanted to clarify a couple things to be sure I have them correct: 

1. With respect to tagging the VNets do I have the format correct as I have been unsuccessful in having the VNet assign itself automatically to a block when the tags have been configured. 

![image](https://user-images.githubusercontent.com/102684496/228049413-40c514a2-81be-4c65-8a2d-74f60ae4da22.png)

2. In playing around with the APIs I have had success with all of them (great by the way) except for the those that are calling for the Resource ID. Again, wanted to confirm that I have this right for example in an the body of the API call I am required to input the vnet resource id found under properties. Would it be in this format: 

![image](https://user-images.githubusercontent.com/102684496/228050641-07faf8c7-fdbe-4546-8dbb-f52350768e28.png)

3. When running the PowerShell script to grab the required API info I keep on getting a 400 error 

![image](https://user-images.githubusercontent.com/102684496/228055192-7bed13d7-0135-4495-9d33-6cadd722fe69.png)

Thanks for taking the time to review and again hats off for a great IPAM solution!

**Comment by DCMattyG** [comment, marked as answer]
Hey there @A-ESDC, thank you so much for reaching out! Answers as follows:

1. The correct TAG is **X-IPAM-RES-ID** (In fact this is also returned in the body of the API response when creating a reservation). You can find more details about this [HERE](https://azure.github.io/ipam/#/api/README?id=sample-api-calls), though I agree we need to add some better documentation around this.
2. Yes, generally we're referring to the Azure **Resource ID**, which would be in the format you alluded to above. If you're having any issues with specific API's, please let me know and we're happy to help you through them either here in the Discussions, or 1:1.
3. When you're running this, are you performing this action from PowerShell on your local machine, or Azure Cloud Shell? Additionally, are you logged in as yourself, or a Service Principal?

Hope that helps, have a wonderful rest of the day!

**Reply by esdcharles** [reply]
Perfect ! Will try tomorrow in the following format (please correct me if I still got it wrong): 
![image](https://user-images.githubusercontent.com/102684496/228056368-c098ebe2-bd3b-4096-a4de-cc7c4da043ba.png)

Thanks

**Reply by DCMattyG** [reply]
This is 100% correct. Please let me know if you have any issues with the TAG association.

**Comment by esdcharles** [comment]
@DCMattyG Just wanted to follow up on this and let you know all of the above has since been resolved. Thanks again!

**Comment by omerzubair** [comment]
hi @DCMattyG , 
i would like to echo @binarybfc i.e its a wonderfull implementation and deploys first time! Thanks for all the hard work. Looking forward for a secure version landing in couple of weeks :) before i show it to my security team.  

i have raised something on tags. if we have 2x cider in same vnet then how to add second tage with same name?
https://github.com/Azure/ipam/discussions/216
